File: 02_draw_line/java/main.java
import java.awt.Color;
import java.util.Random;

public class main
{
    public static void main(String[] args)
    {
	screen s = new screen();
	/*for(int x = 0; x < 600; x++)
	{
	    for(int y = 0; y < 600; y++)
	    {
		int r = (int)((Math.sqrt(Math.pow(x - 300, 2) + Math.pow(y - 300, 2)))/1.655);
		int g = 2 * 256-(int)((Math.sqrt(Math.pow(x - 100, 2) + Math.pow(y - 100, 2)) + Math.sqrt(Math.pow(x - 500, 2) + Math.pow(y - 500, 2)))/2.0);
		int b = (int)((Math.sqrt(Math.pow(x - 100, 2) + Math.pow(y - 500, 2)) + Math.sqrt(Math.pow(x - 500, 2) + Math.pow(y - 100, 2)))/2.0);
		s.setPixel(x, y, new Color(r%256, g%256, b%256));
	    }
	}*/
	Random R = new Random();
	for(int x0 = 50; x0 < 600; x0 = x0 + 100)
	{
	    for(int y0 = 50; y0 < 600; y0 = y0 + 100)
	    {
		for(int x1 = 50; x1 < 600; x1 = x1 + 100)
		{
		    for(int y1 = 50; y1 < 600; y1 = y1 + 100)
		    {
			int r = R.nextInt(256);
			int g = R.nextInt(256);
			int b = R.nextInt(256);
			s.drawLine(x0, y0, x1, y1, new Color(r, g, b));
		    }
		}   
	    }   
	}
	s.saveScreen("pic.ppm");
    }
}

File: 02_draw_line/c/main.c
#include <stdio.h>
#include <stdlib.h>
#include <string.h>
#include <math.h>

#include "ml6.h"
#include "display.h"
#include "draw.h"

void randomizeColor(color*);

int main(int argc, char** argv)
{
  screen s;
  color c;
  double x = 20;
  double y = 250;
  double x2, y2;
  double i = 1;

  srand(time(0));
  
  clear_screen(s);
  
  do
  {
    randomizeColor(&c);
    
    x2 = 250 + (cos(i) * (x - 250) - sin(i) * (y - 250));
    y2 = 250 + (sin(i) * (x - 250) + cos(i) * (y - 250));
    
    draw_line((int)x, (int)y, (int)x2, (int)y2, s, c);
    
    x = x2;
    y = y2;
  }
  while(!(x < 20.1 && x > 19.9 && y < 250.1 && y > 249.9));
  
  if(argc > 1 && !strcmp(argv[1], "-d"))
  {
    display(s);
  }
  else
  {
    save_ppm(s, "pic.ppm");
  }
}

void randomizeColor(color* c)
{
  c->red = rand() % 200 + 56;
  c->blue = rand() % 200 + 56;
  c->green = rand() % 200 + 56;
}
